components/CreatePost.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+
+type NewPost = {
+  titleDescription: string;
+  type: string;
+};
+
+const iniState = {
+  titleDescription: "",
+  type: "Life",
+};
+
+export const CreatePost = () => {
+  const POST_TYPES = ["Life", "Job", "Joke", "Animals", "Food"];
+  const [isHidden, setIsHidden] = useState(true);
+  const [newPost, setNewPost] = useState<NewPost>(iniState);
+
+  const isEmpty = (str: string) => {
+    const length = str.trim().length;
+    return length < 1;
+  };
+
+  const handleToggleCreatePost = () => {
+    setIsHidden((prev) => !prev);
+  };
+
+  const handleOnSubmit = (e: any) => {
+    e.preventDefault();
+    if (isEmpty(newPost.titleDescription)) {
+      return;
+    }
+    handleToggleCreatePost();
+    //POST to DB
+    //axios.post(url, {newPost})
```

Comment:
remove comments, code should be self describing and the main branch should not have any comments